### DISCUSSION TITLE
ERROR: Failed to resolve version

### DISCUSSION THREAD

**Discussion by PsychoPunkSage** [Cyfrin/foundry-full-course-cu#772] [Q&A]
Hi,
I'm trying to use ERC1155 in a project.
But I'm facing an error cause .sol files I'm using are of different versions.
<img width="526" alt="image" src="https://github.com/Cyfrin/foundry-full-course-f23/assets/95116715/d31f4cfa-ffe7-4463-b59e-001bb121e649">
Can Anyone tell me how to resolve this issue?

**Comment by theirrationalone** [comment]
Make sure you have placed imports below `pragma solidity x. X. X... ` if yes then add... Below chunki into your foundry.toml and will automatically download required versions. 


```toml
solc = x.x.x
```

If above doesn't work then try this:

```toml
solc_version = x.x.x
```
Note:
e.g. x.x.x = "0.8.0" Or x.x.x = 0.8.0

**Comment by alfheimrShiven** [comment]
@APrakash-913 your `GameItems1.sol` is using solidity version `^0.6.0` which is incompatible with the `DeployGameItems.s.sol` solidity version of `^0.8.18`. The point to note here is that major solidity release versions like `^0.x` and `^0.y` will not be compatible as they will have breaking changes. Hence bump your solidity version of the `GameItems1.sol` from `^0.6.xx` to `^0.8.xx` and it should work.

Update the following line in your `GameItems1.sol`
`pragma solidity ^0.6.0 ❌`
`pragma solidity ^0.8.18 ✅`
